src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -671,6 +671,10 @@ bool CDSConfig::is_dumping_preimage_static_archive() {
   return _is_dumping_static_archive && CacheDataStore != nullptr && CDSPreimage == nullptr;
 }
 
+bool CDSConfig::is_dumping_preimage_static_archive_with_triggers() {
+  return _is_dumping_static_archive && CacheDataStore != nullptr && CDSPreimage == nullptr && !FLAG_IS_DEFAULT(AOTEndTrainingOnMethodEntry);
```

Comment:
This can be replaced with `is_dumping_preimage_static_archive() && !FLAG_IS_DEFAULT(AOTEndTrainingOnMethodEntry)`